elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,605 @@
+                        provider=self._type,
+                        request_history=request_history)
+
+            # if we are at `/auth/xxxx/login` path, then no further action is needed
+            # (we can use it for login POST)
+            if re.search(r"/auth/.*/login$", redirect_url_obj.path):
+                dex_login_url = redirect_url_obj.geturl()
+            else:
+                # else, we need to be redirected to the actual login page
+                # this GET should redirect us to the `/auth/xxxx/login` path
+                resp = s.get(redirect_url_obj.geturl(), allow_redirects=True)
+                request_history.append({'request_url': redirect_url_obj.geturl(), 'response': resp})
+                if resp.status_code != HTTPStatus.OK:
+                    raise AuthenticationError('Error redirecting to the DEX login page: '
+                                              f'HTTP status code {resp.status_code}.',
+                                              provider=self._type,
+                                              request_history=request_history)
+                # set the login url
+                dex_login_url = resp.url
+
+            ################
+            # Attempt Dex Login
+            ################
+            resp = s.post(
+                dex_login_url,
+                data={"login": username, "password": password},
+                allow_redirects=True
+            )
+            request_history.append({'request_url': dex_login_url, 'response': resp})
+
+            if len(resp.history) == 0:
+                raise AuthenticationError('The LDAP credentials are probably invalid. '
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            # store the session cookies in a "key1=value1; key2=value2" string
+            return "; ".join([f"{c.name}={c.value}" for c in s.cookies])
+
+        # this code should never be reached; raise an error
+        raise AuthenticationError('An implementation problem was detected for LDAP authentication. '
+                                  'Please create an issue.',
+                                  provider=self._type,
+                                  request_history=request_history)
+
+
+class SATokenPathKFPAuthenticator(AbstractKFPAuthenticator):
+
+    _type = SupportedKFPAuthProviders.KF_PIPELINES_SA_TOKEN_PATH.value
+
+    def authenticate(self,
+                     kf_endpoint: str,
+                     runtime_config_name: str) -> Optional[str]:
+
+        request_history = []
+
+        # Verify the API endpoint
+        resp = requests.get(kf_endpoint)
+        request_history.append({'request_url': kf_endpoint, 'response': resp})
+        if resp.status_code != HTTPStatus.OK:
```

Comment:
Why are you checking for HTTP 200 here, there are many other successful HTTP codes?

Reply:
You are right. The main purpose of this test was to validate that the URL is valid. I've updated the code to follow redirects, which should now yield status code 200.